src/gui/plugins/plotting/Plotting.cc

```diff
@@ -0,0 +1,405 @@
+ /*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <ignition/plugin/Register.hh>
+
+#include "Plotting.hh"
+#include <memory>
+
+#include "ignition/gazebo/components/Actor.hh"
+#include "ignition/gazebo/components/AngularAcceleration.hh"
+#include "ignition/gazebo/components/AngularVelocity.hh"
+#include "ignition/gazebo/components/CastShadows.hh"
+#include "ignition/gazebo/components/ChildLinkName.hh"
+#include "ignition/gazebo/components/Collision.hh"
+#include "ignition/gazebo/components/Factory.hh"
+#include "ignition/gazebo/components/Gravity.hh"
+#include "ignition/gazebo/components/Joint.hh"
+#include "ignition/gazebo/components/Level.hh"
+#include "ignition/gazebo/components/Light.hh"
+#include "ignition/gazebo/components/LinearAcceleration.hh"
+#include "ignition/gazebo/components/LinearVelocity.hh"
+#include "ignition/gazebo/components/LinearVelocitySeed.hh"
+#include "ignition/gazebo/components/Link.hh"
+#include "ignition/gazebo/components/MagneticField.hh"
+#include "ignition/gazebo/components/Model.hh"
+#include "ignition/gazebo/components/Name.hh"
+#include "ignition/gazebo/components/ParentEntity.hh"
+#include "ignition/gazebo/components/ParentLinkName.hh"
+#include "ignition/gazebo/components/Performer.hh"
+#include "ignition/gazebo/components/PerformerAffinity.hh"
+#include "ignition/gazebo/components/Pose.hh"
+#include "ignition/gazebo/components/PoseCmd.hh"
+#include "ignition/gazebo/components/Sensor.hh"
+#include "ignition/gazebo/components/Static.hh"
+#include "ignition/gazebo/components/Visual.hh"
+#include "ignition/gazebo/components/WindMode.hh"
+#include "ignition/gazebo/components/World.hh"
+#include "ignition/gazebo/EntityComponentManager.hh"
+
+namespace ignition::gazebo
+{
+  class PlottingPrivate
+  {
+    /// \brief Interface to communicate with Qml
+    public: ignition::gui::PlottingInterface *plottingIface;
+
+    /// \brief registered components for plotting
+    /// map key: string contains EntityID + "," + ComponentID
+    public: std::map<std::string, PlotComponent*> components;
```

Comment:
Do the `PlotComponent`s need to be pointers or can they just be objects? Then we don't need to worry about deleting them. I think that this memory isn't being cleared right now.

Reply:
Yeah, i add them at run time and i PlotData class dosn't have Copy Constructor to assign local variables to it without dynamically allocating (will require another PR to ign-gui code)

anyway, i handled deleting them